packages/safe-tools-client/app/components/future-payments-list/index.gts

```diff
@@ -1,49 +1,110 @@
 import Component from '@glimmer/component';
 import BoxelActionContainer from '@cardstack/boxel/components/boxel/action-container';
-import WalletService from '@cardstack/safe-tools-client/services/wallet';
 import NetworkService from '@cardstack/safe-tools-client/services/network';
 import HubAuthenticationService from '@cardstack/safe-tools-client/services/hub-authentication';
 import ScheduledPaymentsService from '@cardstack/safe-tools-client/services/scheduled-payments';
-import TokensService from '@cardstack/safe-tools-client/services/tokens';
 import { inject as service } from '@ember/service';
 import { on } from '@ember/modifier';
-import { action } from '@ember/object';
+import { use, resource } from 'ember-resources';
+import { TrackedObject } from 'tracked-built-ins';
+import { taskFor } from 'ember-concurrency-ts';
+import { task, TaskGenerator } from 'ember-concurrency';
 import lt from 'ember-truth-helpers/helpers/lt';
+import TimeBracket from './time-bracket';
+import { ScheduledPayment } from '@cardstack/safe-tools-client/services/scheduled-payments';
+import { addHours, addMonths, lastDayOfMonth } from 'date-fns';
 
 import './index.css';
 
-export default class FuturePaymentsList extends Component {
+interface Signature {
+  Element: HTMLElement;
+  Args: {
+    onDepositClick: () => void;
+  }
+}
+
+export default class FuturePaymentsList extends Component<Signature> {
   @service declare hubAuthentication: HubAuthenticationService;
   @service declare network: NetworkService;
   @service declare scheduledPayments: ScheduledPaymentsService;
-  @service declare tokens: TokensService;
-  @service declare wallet: WalletService;
 
-  get futurePayments() {
-    return [] //TODO: fetch future payments
+  get nextHour(): ScheduledPayment[] {
+    if (!this.futurePaymentsResource.value) return [];
+    return this.futurePaymentsResource.value.filter((s: ScheduledPayment) => s.payAt >= addHours(new Date(), 1) && s.payAt <= addHours(new Date(), 2));
```

Comment:
Rather than perform the same date math every time the the filter executes, calculate "oneHourFromNow" on the line before this one.